plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2MavenPathParser.java

```diff
@@ -106,23 +108,41 @@ else if (str.endsWith("." + HashType.MD5.getExt())) {
           tail = str.substring(nTailPos);
         }
         else {
-          StringBuilder snapshotBuildNumber = new StringBuilder(version);
-          snapshotBuildNumber
-              .append(str.substring(vSnapshotStart + version.length(), vSnapshotStart + version.length() + 8));
+          final StringBuilder snapshotTimestampedVersion = new StringBuilder(version);
+          snapshotTimestampedVersion.append(
+              str.substring(vSnapshotStart + version.length(), vSnapshotStart + version.length() + 7)
```

Comment:
Yes, as @kellyrob99 pointed out, the intention was to leave out the "-" (dash) from the date:
https://github.com/sonatype/nexus-oss/pull/1257#discussion_r29607820